chpl/chpl-auth/src/main/java/gov/healthit/chpl/auth/manager/impl/UserManagerImpl.java

```diff
@@ -275,16 +327,17 @@ public String getEncodedPassword(final UserDTO user) throws UserRetrievalExcepti
         return userDAO.getEncodedPassword(user);
     }
 
-
     @Override
     public Set<UserPermissionDTO> getGrantedPermissionsForUser(final UserDTO user) {
         return securedUserManager.getGrantedPermissionsForUser(user);
     }
 
-
     @Override
     public UserDTO getByName(final String userName) throws UserRetrievalException {
-        return securedUserManager.getBySubjectName(userName);
+        loginAdminUser();
```

Comment:
Where is this being called from in the new code that is requiring a user to be logged in when it's called? Could the place that's calling it just go to the DAO instead of the manager?

Reply:
Its being called in the AutheticationController.resetPassword method. Because people are resetting passwords they probably aren't logged in (at least I don't think that's possible). So we need to login temporarily so that we can get past the authentication in the securedUserManager class. I never though about calling DAO's directly though....

Reply:
It just seems like this method is one that could be (is?) called from elsewhere in the application and if it was the behavior could be odd. Like, what happens if you're logged in as someone (not admin) and do some action that calls this code (I don't know if that is currently possible, but maybe)? Are you then logged out if that happens? I think this needs to be changed around in some way so that the login and logout lines are not a part of this method. Maybe that's calling the DAO from the controller - which we don't normally do but maybe it's ok in this case? - or maybe that's adding some other non-secured manager method.